task_opts.go

```diff
@@ -51,3 +51,15 @@ func WithProcessKill(ctx context.Context, p Process) error {
 	<-s
 	return nil
 }
+
+type KillInfo struct {
+	// All kills all processes inside the task
+	All bool
+}
+
+type KillOpts func(context.Context, Process, *KillInfo) error
```

Comment:
nit: Could the options just take in the `KillRequest` type?

Reply:
No, we have been going away from exposing the protos in the client interfaces

Reply:
:ok_hand: